runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,411 @@
+package org.corfudb.runtime;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.protobuf.Message;
+import com.google.protobuf.Timestamp;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.LogReplication.ClientDestinationInfoKey;
+import org.corfudb.runtime.LogReplication.ClientRegistrationId;
+import org.corfudb.runtime.LogReplication.ClientRegistrationInfo;
+import org.corfudb.runtime.LogReplication.DestinationInfoVal;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.Table;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.exceptions.TransactionAbortedException;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+import org.corfudb.util.retry.ExponentialBackoffRetry;
+import org.corfudb.util.retry.IRetry;
+import org.corfudb.util.retry.RetryNeededException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import static org.corfudb.runtime.view.TableRegistry.CORFU_SYSTEM_NAMESPACE;
+
+/**
+ * A client for log replication
+ */
+@Slf4j
+public class LogReplicationLogicalGroupClient {
+
+    /**
+     * Key: ClientRegistrationId
+     * Value: ClientRegistrationInfo
+     */
+    public static final String LR_REGISTRATION_TABLE_NAME = "LogReplicationRegistrationTable";
+
+    /**
+     * Key: ClientDestinationInfoKey
+     * Value: DestinationInfoVal
+     */
+    public static final String LR_MODEL_METADATA_TABLE_NAME = "LogReplicationModelMetadataTable";
+
+    private static final String modelEnum = "LOGICAL_GROUPS";
```

Comment:
I notice all the usages of this string is to set the model field in ClientDestinationInfoKey, then I think we can just do something like
```
    private static final ReplicationModel model = ReplicationModel.LOGICAL_GROUPS;
```